### DISCUSSION TITLE
TWG: discussion on defining process for Continuous Testing on Foorprint/performance/low power

### DISCUSSION THREAD

**Discussion by hakehuang** [zephyrproject-rtos/zephyr#33779] [Ideas] [answered]
For better qulatiy track, below items need to be defined:

1. How to track footprint, and on which boards
curently we have this [application](https://github.com/zephyrproject-rtos/zephyr/wiki/%5BHOW-TO%5D-Generate-a-binary-footprint-for-a-basic-Zephyr-application), maybe we need create a list of boards to track for each PR.

2.  How to track performance, and on which boards
currently, we have this [application](https://github.com/zephyrproject-rtos/zephyr/wiki/%5BHOW-TO%5D-Generate-benchmarks-metrics)
we need define a list of boards to track and more system level applications

3. Low power
same with 1, 2.

**Comment by robwoolley** [comment, marked as answer]
[Here](https://docs.google.com/presentation/d/1CNMtN-A4B7K64ncQ52MIDIHHQpITLfMuX0bv-OQnUj0/edit?usp=sharing) are the slides I presented today

We used “intercept” from Build EAR to gather the compiler and linker information from another RTOS: https://github.com/rizsotto/Bear

We did some post-processing in JSON to gather the relevant pieces of information and visualized the data using Gephi: https://gephi.org/

The compiler graph was generated using the clang -ftime-tracing flag and visualized using Google Chrome Tracing.  There is a useful write-up here: https://aras-p.info/blog/2019/01/16/time-trace-timeline-flame-chart-profiler-for-Clang/   this Build Analyzer may be interesting to this group as well: https://aras-p.info/blog/2019/09/28/Clang-Build-Analyzer/
